src/docs/tiered-storage.dox

```diff
@@ -0,0 +1,82 @@
+name on disk, in cloud storage and in metadata.
+
+When a tiered btree is created, there is a single file associated with it,
+it is given object number 1. This is the active file, and it only resides
+on disk. This is always true for an active file. When a file ceases
+to be an active file, writes to it will no longer be permitted.  At that
+point, such a file can be copied to cloud storage.  How this transition
+to a new active file occurs is the subject of the next section.
+
+@section flush_tier The flush_tier operation
+
+A fundamental operation for a tiered table is \a flush_tier, this occurs
+as a result of a call to WT_SESSION::checkpoint with a configuration
+string that includes \c "flush_tier=(enabled=true)".  Because flush_tier
+is a part of checkpoint, it is a system-wide operation. Thus, flush_tier
+allows the data for all tiered btrees to move from the local file system
+to object storage.
+
+The flush_tier checkpoint does all the operations normally associated with
+a checkpoint, and in addition:
+
+ -# identifies any tiered btrees that have changes since the previous checkpoint
+ -# for each such modified btree:
+     -# eviction is temporarily disabled on this btree, while waiting for any writes to the active file to drain
+     -# a new empty file is created named with the next object number, this will become the table's active file
+     -# switch the table's active file
+     -# enable eviction on this btree
+     -# queue the old active file to be written to object storage in the background
+ -# when every modified btree has finished switching in this way, the flush_tier part of the checkpoint is finished.
+    Writes to object storage are likely to be incomplete at this stage, they are completely asynchronous.
+
+As each object storage write completes, the local file that was the source of that write will be removed.
+The removal is coordinated with any active readers of that file.  Since the file is no longer an active file,
+by definition there can be no outstanding writers to the file, only readers.
+
+A user of the WiredTiger API does not need to be aware of this coordination. Cursors may remain open on a tiered table,
+and reads, writes and transactions may occur as usual, unaffected by the machinery of the flush_tier. Generally the
+changes to an application are minor:
+
+  - indicate on a ::wiredtiger_open call that tiered storage is being used, and specify
+    the storage provider and bucket name.
+  - for tables that should not participate in tiered storage, or have a different storage provider or bucket,
+    indicate on the WT_SESSION::create calls for that table.
+  - modify the application's checkpoint thread to include \c "flush_tier=(enable=true)" on calls to WT_SESSION::checkpoint.
+    \c flush_tier need not be specified on every checkpoint call; typically the cadence of \c flush_tier is much less than
+    the cadence of ordinary checkpoints.
```

Comment:
Reading this, it implies (or requires) that using tiered storage also requires using an application checkpoint thread. I.e. I don't think we provide any mechanism to flush tier via the WT internal checkpoint server. Probably useful to state that for now even if we could (and should) add it later.

Reply:
added a couple sentences to this effect.